### ISSUE TITLE
`reportUnusedParameter` false negative on overridden method when the unused argument isn't present in the base method

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#673]
Code sample in [basedpyright playground](https://basedpyright.com/?typeCheckingMode=all&code=GYJw9gtgBALgngBwJYDsDmUkQWEMpgBuApiCEgCbECwAUHXQMYA2AhgM7tQBiYYAXHSjCoVYFABGrEAAp2xZsAA0UQq2YBXYv0woYASh0A6E1ADEUUuBA6QxHHgCqKDfIoAFaawjEYpBrQsHFwAQqwAXjK8YIZCIgACRFaUNLQiosTiUrLyiipqmtq6MPnqWgBMOqj4AD5QAHJgKMRQALwNTcSG5lAoYJZkuHHpwgjkejIFWvpAA)

```python
from typing import override


class Foo:
    def bar(self, value: int): ... # error: reportUnusedParameter

class Baz(Foo):
    @override
    def bar(self, value: int, value2: int | None = None): # no error
        print(value)
```
